packages/private-apis/src/implementation.ts

```diff
@@ -170,17 +172,19 @@ function lock( object: Record< symbol, WeakKey >, privateData: unknown ) {
  * @param object The object to unlock the private data from.
  * @return The private data bound to the object.
  */
-function unlock( object: Record< symbol, WeakKey > ) {
+function unlock< T >( object: T ): T {
```

Comment:
This looks wrong to me personally, unlock doesn't return the same "type" of the object you give it.

Reply:
It was as per the suggestion by @jsnajdr in https://github.com/WordPress/gutenberg/pull/66667#discussion_r1825958215

I think the logic inside `lock` and `unlock` should be alien to the consumer code and it should/need not know what it internally does, because we expect them to use `lock` `unlock` together, not in isolation.

Reply:
But this is different from @jsnajdr's suggestion, the type of the argument and the returned value don't have the same types there.

Reply:
Then, in that case, there will be no inference from TypeScript, and the consumer will always have to pass the type as the generic to ensure that the return type is not always `unknown,` which can be annoying.

Reply:
I know but if we return the same type, it's wrong too because we'll either have locked items missing from the returned value if the returned object contains the original one (which happens for store), or we'll have a completely wrong return type.

I see things closer to this but I'm not sure how doable:

```
 lock( o: X, privateApis:  Y ): XwithLockedY
 unlock( XwithLockedY ):  X & Y
```

Reply:
Thanks, I have updated the signature to `unlock< T = any >( object: unknown ): T` to use `any` as default to avoid breaking existing types.

Reply:
Thanks, I'll let others, more experts with TS chime in. cc @jsnajdr @sirreal if needed.